Automattic/wp-calypso
# JP Manage: 221 - add unused license notice on upsell pages

Resolves Automattic/jetpack-manage#221 - add unused license notice on upsell pages

## Proposed Changes

Currently on the My Sites upsell pages we prompt users to upgrade even if they already have an unassigned license that might add the feature. This PR adds a notice that checks existing licenses to see if there's one that fits the bill, thanks to the new `productHasFeatureType` function.

Clicking the X will dismiss the notice (for that page load). Clicking "View licenses" will take one to the unassigned licenses page. Either click will fire a Tracks event.

Caveats: we only check the first 50 unassigned licenses associated with the selected key. This is good enough for an MVP; a more robust check should be done on the backend (low-priority issue forthcoming).

## Testing Instructions

1. Make sure you have an unassigned Backup, Scan, or Search license on your selected key.
2. Go to an upsell page.

In staging/`trunk`, you'll only see the upsell:
![image](https://github.com/Automattic/wp-calypso/assets/32492176/52000591-4b52-49d4-b78d-174f27ad14ac)

In the `fix/jp-manage/221-add_existing_license_notice_on_upsell_pages` branch you'll also see a notice:

![image](https://github.com/Automattic/wp-calypso/assets/32492176/1935e72e-2f43-47ee-a140-78705a871a2e)

## Pre-merge Checklist
